tests/integration/relationships/nested-relationship-test.js

```diff
@@ -35,6 +37,19 @@ module('integration/relationships/nested_relationships_test - Nested relationshi
       adapter: DS.JSONAPIAdapter
     });
 
+    env.registry.register('adapter:middle-ager', DS.JSONAPIAdapter.extend({
+      findHasMany (store, snapshot, url, relationship) {
+        let data = [];
+        if (url === '/middle-agers/1/kids') {
+          data.push({
+            id: 1,
+            type: 'kid'
+          });
+        }
+        return resolve({ data });
+      }
+    }));
```

Comment:
I am not sure about this approach for fixing the failing test, as it seems this introduces a change in the semantics: before no link was called, now we do when we get the `kids` relationship on line [138/153](https://github.com/emberjs/data/pull/5257/files#diff-e4e8cc9dae6301ff76229fabdb9e3434R153):

```js
assert.ok(middleAger.get('kids').includes(kid));
```

This essentially boils down to if a relationship should be considered loaded when link & data is provided 😔 .

For this PR it might be best to change the assertion so we check the content of the relationship in a side-effect free way:

```js
assert.ok(middleAger.hasMany('kids').value().includes(kid));
```